### ISSUE TITLE
Typescript union of `number` generates `string` enum in swagger

### ISSUE THREAD

**Issue by tsimbalar** [lukeautry/tsoa#1068]
## Sorting

- **I'm submitting a ...**

  - [x] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

Given a type defined as 

```ts
export interface MyAnswersRequest {
  readonly answers: {
    readonly answer1: AnswerValue;
  };
}

export type AnswerValue = 0 | 1 | 2 | 3;
```

I would expect it to generate OpenApi.yaml schemas such as : 

```yaml
        AnswerValue:
            type: number
            enum:
                - 0
                - 1
                - 2
                - 3
        MyAnswersRequest:
            properties:
                answers:
                    properties:
                        answer1:
                            $ref: '#/components/schemas/AnswerValue'
                    required:
                        - answer1
                    type: object
            required:
                - answers
            type: object
            additionalProperties: false
```

## Current Behavior

Instead, the generated enum members are generated as `string`s (with single quotes)

```yaml
        AnswerValue:
            type: number
            enum:
                - '0'
                - '1'
                - '2'
                - '3'
```

Our OpenApi linter (Spectral) complains (and it seems right) that even though the type of this enum is `number` , the listed acceptable values are `string`s 

Note that this only seems to impact the generated swagger file, but validation of incoming requests work properly.

## Steps to Reproduce

1. 2. 3. 4.

## Context (Environment)

Version of the library: `"@tsoa/runtime": "3.9.0",` / `"@tsoa/cli": "3.9.0",`
Version of NodeJS: `v14.16.0`

- Confirm you were using yarn not npm: [ ]

## Breaking change?

This might be a breaking change for people who rely on unions of `number` ... but this would only impact the generated swagger file, not the runtime behavior.

**Comment by WoH** [comment]
I suspect this happens when we transform to yaml. Does a spec JSON have numbers?

**Comment by tsimbalar** [comment]
I just gave it a try by switching `yaml` to `false` in `tsoa.json`.

The generated `swagger.json` looks like this : 


```json
"AnswerValue": {
				"type": "number",
				"enum": [
					"0",
					"1",
					"2",
					"3"
				]
			},
```

so the `enum` members are generated as `string`s , not numbers in the JSON as well.

**Comment by tsimbalar** [comment]
I can confirm this is now fixed as of `tsoa/cli` v3.14.1 
probably thanks to changes in : 
- #1117
